### DISCUSSION TITLE
Docker setup very slow on digital ocean droplet

### DISCUSSION THREAD

**Discussion by ethanitovitch** [cookiecutter/cookiecutter-django#4682] [Q&A]
Hi all, 
I setup cookiecutter-django using docker.
I then deployed the app on a digital ocean droplet.
Here are the current specs of the machine: 2 GB Memory / 2 Intel vCPUs / 90 GB Disk / SFO3 - Ubuntu 22.04 (LTS) x64

I've deployed many apps on droplets with the same specs that ran much faster without docker so thinking that's the issue.
Also I've stripped the docker compose file so that it's just traffik, redis & django.

Any ideas what might be going wrong? The site has 0 traffic.

Also please let me know how I can provide more info I know this is extremely vague.

Thanks!

**Comment by dacx** [comment]
Is the droplet running out of resources, or are your requests slow for another reason? Did you look at htop or another monitor?

**Reply by ethanitovitch** [reply]
So we've since switched our deployment to standard django project on a digital ocean app. But there are the stats from docker stats and free -h

here are the specs of my machine: 2 GB Memory / 2 Intel vCPUs / 90 GB Disk / SFO3 - Ubuntu 22.04 (LTS) x64

Here is docker stats
CONTAINER ID   NAME               CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O     PIDS
734f51a60a7d   app_traefik_1   0.00%     22.41MiB / 1.918GiB   1.14%     1.49MB / 1.59MB   10.7MB / 0B   10
6b63ffaa3e33   app_django_1    0.01%     395.3MiB / 1.918GiB   20.13%    455kB / 1.72MB    16.1MB / 0B   5
7b63ee60bb79   app_redis_1     0.15%     4.914MiB / 1.918GiB   0.25%     28.7kB / 21.8kB   6.52MB / 0B   5

free -h
               total        used        free      shared  buff/cache   available
Mem:           1.9Gi       1.1Gi       301Mi       4.0Mi       494Mi       622Mi
Swap:             0B          0B          0B

**Reply by foarsitter** [reply]
Where is your database?

**Reply by ethanitovitch** [reply]
We used managed postgreql on digital ocean.
It was in the same region as our droplet.

**Reply by foarsitter** [reply]
Docker shoudn't be the problem. Can you share your Django settings and requirement.txt? It may be a middleware or some other package.

**Reply by ethanitovitch** [reply]
hmm but now we're deployed on a digital ocean app without docker and it's a lot faster

**Reply by ethanitovitch** [reply]
hmm but now we're deployed on a digital ocean app without docker and it's a lot faster

**Reply by ethanitovitch** [reply]
Guessing it's some connection with database, maybe sentry?
